test/it/parser/src/main/resources/case/dml/select-expression.xml

```diff
@@ -3178,4 +3178,48 @@
             </expression-projection>
         </projections>
     </select>
+
+    <select sql-case-id="select_with_case_when_group_by_order_by">
```

Comment:
Does this PR not involve any changes to G4 or visit logic? In other words, the SQL on the issue is already supported, in this case, we just need to leave a message on the issue, and the community volunteer will close the issue.